hazelcast/src/main/java/com/hazelcast/map/impl/record/RecordReaderWriter.java

```diff
@@ -99,9 +127,11 @@ void writeRecord(ObjectDataOutput out, Record record, Data dataValue,
             writeData(out, dataValue);
             out.writeInt(record.getVersion());
             out.writeInt(record.getRawLastAccessTime());
-            out.writeInt(expiryMetadata.getRawTtl());
-            out.writeInt(expiryMetadata.getRawMaxIdle());
-            out.writeInt(expiryMetadata.getRawExpirationTime());
+            // RU_COMPAT_4_2
+            Version version = out.getVersion();
+            if (!version.isGreaterOrEqual(Versions.V5_0)) {
+                expiryMetadata.write(out);
+            }
```

Comment:
I think if we completely move these expiryMetadata serde staff into `Records#writeExpiry(ObjectDataOutput , ExpiryMetadata)` and `Records#readExpiry(ObjectDataInput)`, it would be better for readability. Since we're doing this staff in two different places for `versions >= 5.0` and `versions < 5.0` now, it's a little difficult to follow.

Reply:
I accept it is a bit hard to read now but it will be better in 5.1 after deleting compatibility paths. The reason that we can't put all expiry serde in `writeExpiry` is different type of records we have. In those records order of serde fields are different, so for now i will not change this since this is a temporary state.